fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -1818,4 +1824,51 @@ public boolean hasContractTerminationTransaction() {
         return getLoanTransactions().stream().anyMatch(t -> t.isContractTermination() && t.isNotReversed());
     }
 
+    public BigDecimal calculateAvailableDisbursementAmountWithOverApplied() {
```

Comment:
Would you mind to move this calculation into a Calculator component or the DelinquencyReadPlatformService similarly as `calculateAvailableDisbursementAmount(loan)` does? We would like to offload the JPA entities, not introducing further complex calculations.